### DISCUSSION TITLE
Reusing common Interactions?

### DISCUSSION THREAD

**Discussion by ibrahim324** [reactiveui/ReactiveUI#3670] [Q&A] [answered]
I have a ViewModel with an Interaction<Unit, string>, which is responsible of opening a FolderDialog. I "invoke" this Interaction via a Command bound to my view. 
```
private async Task<string> getDirectory()
{
    var dir = await GetDirectoryInteraction.Handle(Unit.Default);
    return string.IsNullOrWhiteSpace(dir) 
        ? string.Empty 
        : dir;
}
```

I have two commands that i want to use this interaction with: 
`GetDirectoryCommand = ReactiveCommand.CreateFromTask(getDirectory);`
and
```
MoveDocumentsCommand = ReactiveCommand.CreateFromTask(async () =>
{
    var dir = await getDirectory();
    if (Directory.Exists(dir))
    {
        foreach (var item in this.documents.Items)
        {
....
```

But i realized that the getDirectory-Interaction returns the same string when i call it in the second command!

**Comment by anaisbetts** [comment, marked as answer]
Interactions end up "walking" up the stack of registered interactions - if you want to reuse a common one, you can declare it in a common parent View of both elements
